operator/v1/types_csi_cluster_driver.go

```diff
@@ -168,6 +168,61 @@ type AWSCSIDriverConfigSpec struct {
 	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b|aws-iso-e|aws-iso-f):kms:[a-z0-9-]+:[0-9]{12}:(key|alias)\/.*$`
 	// +optional
 	KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+
+	// efsVolumeMetrics sets the configuration for collecting metrics from EFS volumes used by the EFS CSI Driver.
+	// +optional
+	EFSVolumeMetrics *AWSEFSVolumeMetrics `json:"efsVolumeMetrics,omitempty"`
+}
+
+// AWSEFSVolumeMetricsState defines if the collection of volume metrics in the AWS EFS CSI Driver is enabled or disabled.
+// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+// +kubebuilder:validation:Enum=Enabled;Disabled
+type AWSEFSVolumeMetricsState string
+
+const (
+	// AWSEFSVolumeMetricsEnabled defines that the collection of volume metrics in the AWS EFS CSI Driver is enabled.
+	AWSEFSVolumeMetricsEnabled AWSEFSVolumeMetricsState = "Enabled"
+
+	// AWSEFSVolumeMetricsDisabled defines that the collection of volume metrics in the AWS EFS CSI Driver is disabled.
+	AWSEFSVolumeMetricsDisabled AWSEFSVolumeMetricsState = "Disabled"
+)
+
+// AWSEFSVolumeMetrics defines the configuration for Volume Metrics in the EFS CSI Driver.
+type AWSEFSVolumeMetrics struct {
+	// state defines the state of metric collection in the AWS EFS CSI Driver.
+	// The default value is Disabled.
+	// +openshift:enable:FeatureGate=AWSEFSDriverVolumeMetrics
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:=Enabled;Disabled
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default:=Disabled
+	// +default="Disabled"
+	State AWSEFSVolumeMetricsState `json:"state,omitempty"`
```

Comment:
Technically, we don't need the `State` field if `EFSVolumeMetrics` above is a pointer. The presence or absence of the pointer itself could indicate whether the feature is enabled. WDYT?

Reply:
Great suggestion, thanks!